transform-plugins/src/test/java/co/cask/hydrator/plugin/XMLParserTest.java

```diff
@@ -0,0 +1,338 @@
+import co.cask.cdap.etl.api.Transform;
+import co.cask.cdap.etl.batch.mapreduce.ETLMapReduce;
+import co.cask.cdap.etl.mock.batch.MockSink;
+import co.cask.cdap.etl.mock.batch.MockSource;
+import co.cask.cdap.etl.proto.v2.ETLBatchConfig;
+import co.cask.cdap.etl.proto.v2.ETLPlugin;
+import co.cask.cdap.etl.proto.v2.ETLStage;
+import co.cask.cdap.proto.artifact.AppRequest;
+import co.cask.cdap.proto.id.ApplicationId;
+import co.cask.cdap.proto.id.NamespaceId;
+import co.cask.cdap.test.ApplicationManager;
+import co.cask.cdap.test.DataSetManager;
+import co.cask.cdap.test.MapReduceManager;
+import co.cask.hydrator.common.test.MockEmitter;
+import co.cask.hydrator.common.test.MockTransformContext;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.junit.Assert;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+public class XMLParserTest extends TransformPluginsTestBase {
+  private static final Schema INPUT = Schema.recordOf("input", Schema.Field.of("offset", Schema.of(Schema.Type.INT)),
+                                                      Schema.Field.of("body", Schema.of(Schema.Type.STRING)));
+
+  private static final Logger LOG = LoggerFactory.getLogger(XMLParserTest.class);
+
+  @Test
+  public void testInvalidConfig() throws Exception {
+
+    String inputTable = "input-xpath-with-multiple-elements";
+    ETLStage source = new ETLStage("source", MockSource.getPlugin(inputTable));
+    Map<String, String> sourceProperties = new ImmutableMap.Builder<String, String>()
+      .put("input", "body")
+      .put("encoding", "UTF-8")
+      .put("xpathMappings", "category://book/@category,title://book/title,year:/bookstore/book[price>35.00]/year," +
+        "price:/bookstore/book[price>35.00]/price,subcategory://book/subcategory")
+      .put("fieldTypeMapping", "category:,title:string,price:double,year:int,subcategory:string")
+      .put("processOnError", "Exit on error")
+      .build();
+
+    ETLStage transform = new ETLStage("transform",
+                                      new ETLPlugin("XMLParser", Transform.PLUGIN_TYPE, sourceProperties, null));
+    String sinkTable = "output-xpath-with-multiple-elements";
+
+    ETLStage sink = new ETLStage("sink", MockSink.getPlugin(sinkTable));
+    ETLBatchConfig etlConfig = ETLBatchConfig.builder("* * * * *")
+      .addStage(source)
+      .addStage(transform)
+      .addStage(sink)
+      .addConnection(source.getName(), transform.getName())
+      .addConnection(transform.getName(), sink.getName())
+      .build();
+
+    AppRequest<ETLBatchConfig> appRequest = new AppRequest<>(ETLBATCH_ARTIFACT, etlConfig);
+    ApplicationId appId = NamespaceId.DEFAULT.app("XMLReaderTest");
+    try {
+      deployApplication(appId.toId(), appRequest);
+      Assert.fail();
+    } catch (IllegalStateException e) {
```

Comment:
The error message in this case is a generic message "Expected 200 OK, got 400 Bad Request. Failed to deploy app".Not sure if asserting this is correct.
Changed test case to directly check the exception received from configurePipeline()